src/shell-app.c

```diff
@@ -1095,8 +1128,11 @@ _shell_app_add_window (ShellApp        *app,
 
   if (shell_app_is_interesting_window (window))
     app->running_state->interesting_windows++;
-  else if (shell_window_tracker_is_speedwagon_window (window))
+
+  if (shell_window_tracker_is_speedwagon_window (window)) {
     app->running_state->speedwagon_windows++;
+    app->showing_splash = TRUE;
```

Comment:
Why can't you check `speedwagon_windows > 0` instead of adding a new state?

Reply:
Because by the time `shell_app_sync_running_state()` is called, from `_shell_app_remove_windo()`, `speedwagon_windows` will be `0` again.

That said, it's true this looks a bit awkward, I'll look for a better way to achieve a similar result, if I can find one.